src/mixer.rs

```diff
@@ -212,17 +222,23 @@ impl Mixer {
         let mut i = 0;
 
         while let Some(sound) = self.mixer_state.get_mut(i) {
+            let pitch = sound.pitch;
             let volume = sound.volume;
             let mut remainder = buffer.len();
 
             loop {
```

Comment:
I am not quite sure if this is enough - from what I understand, without interpolation+some filtering, pitch by simple re-sampling with nearest neighbor scaling does not sound very good. 

Also I am not sure if effects belong to the `fill_audio_buffer` function

Reply:
added linear interpolation between sample entries. I don't think it makes much of a difference.